src/lambdaconnect_model/scoping.clj

```diff
@@ -328,29 +310,125 @@
+                                        (into {}))))
+                            (reduce merge))
+        queries (scoping-step
+                 snapshot
+                 #{(:db/id user)}
+                 entities-by-name
+                 {}
+                 #{:user}
+                 relevant-rules
+                 {:user {:dependencies #{} :rules []}})
+        filtered-queries (into {} (filter (fn [[tag _]] (contains? tags tag)) queries))]
+    (->> filtered-queries
+         (pmap (fn [x] (apply (partial execute-query config) x)))
+         (into {}))))
+
+(defn scope-selected-tags
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, a set of desired tags and push?.
+  A typical invocation looks like this: 
+  (scope-selected-tags config (d/db db/conn) user entities-by-name validated-scope desired-tags #{:RARestaurant.ofOwner})))
+  Returns a map with db ids, something like:
+  {:RAOwner.me #{11122, 1222} :user #{2312312}}
+  "
+  [config snapshot user entities-by-name scoping-defintion tags push?]
+  (let [relevant-rules (->> scoping-defintion
+                            (filter (fn [[_ description]]
+                                      (and (:constraint description)
+                                           (or (not push?)
+                                               (-> description :permissions :create)
+                                               (-> description :permissions :modify)
+                                               (-> description :permissions :include-in-push)))))
+                            (map (fn [[tag description]] [tag (:constraint description)]))
+                            (into {}))
+        queries (scoping-step
+                 snapshot
+                 #{(:db/id user)}
+                 entities-by-name
+                 {}
+                 #{:user}
+                 relevant-rules
+                 {:user {:dependencies #{} :rules []}})
+        filtered-queries (into {} (filter (fn [[tag _]] (contains? tags tag)) queries))]
+    (->> filtered-queries
+         (pmap (fn [x] (apply (partial execute-query config) x)))
+         (into {}))))
+
+(defn get-scoping-queries
+  "Takes entities-by-name, parsed (and validated) EDN of scoping rules, a set of desired tags and push?.
+   A typical invocation looks like this: 
+   (get-scoping-queries entities-by-name validated-scope desired-tags #{:RARestaurant.ofEmployee :RARestaurant.ofOwner})))
+   Returns a map with tags and queries generated to obtain entities for indicated user:
+  {:RAEmployee.ofOwner
+  [:find
+  ?RAEmployee-ofOwner
+  :in
+  $
+  [?user ...]
+  :where
+  [?user :app/uuid ?G__33023]
+  [(= ?G__33024 ?G__33023)]
+  [?RAOwner-me :RAOwner/internalUserId ?G__33024]
+  [?RAEmployee-ofOwner :RAEmployee/owner ?RAOwner-me]]}
+   ...
+  "
+  [entities-by-name scoping-defintion tags push?]
+  (let [user {:db/id 1111111}
```

Comment:
jak to ma działać? czy to powinno się tu znaleźć?

Reply:
tak, ogólnie user i snapshot ma znacznie tylko dla wykonywania query, same queries budowane są w oparciu o entites by name i scoping więc de facto są tutaj niepotrzebne ale trzeba je zmockować bo scoping-steps i query-for-rule wymagja ich (używją ich tylko w charakterze załącznika do rezulatatu)

zeby się tego pozbyć możnaby przepisac trochę to query for rule, żeby nie wymagało tych danych i dodać je samemu do queries przed wywołaniem execute-query

Reply:
Okej rozumiem, zaskoczyła mnie ta wartość `{:db/id 1111111}`.
To może dobrze byłoby to przepisać tak, żeby funkcje `query-for-rule` i `scoping-step` nie przyjmowały parametru `snapshot` i `user` albo żeby ta funkcja normalnie przyjmowała snapshot i usera i zwracała query razem z podanymi parametrami snapshot i user (i wtedy będziemy wykonywać query przy pomocy `(apply d/q result)`).

Reply:
Okej, rozumiem. To w takim razie może zamiast {:db/id 11111} i {} można podawać `nile`? Jakoś razi mnie ta wartość `11111`. No i niezależnie od tego przydałby się jakiś komentarz na temat tych wartości user i snapshot.